pkg/operator/controller/controller.go

```diff
@@ -192,9 +192,22 @@ func (r *ReconcileCDI) SetController(controller controller.Controller) {
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileCDI) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
-	operatorVersion := r.namespacedArgs.OperatorVersion
+	if request.NamespacedName.Name == "cdis.cdi.kubevirt.io" {
+		return r.ReconcileCDICRD(request)
```

Comment:
I think it will be better to have a singe Reconcile function.  The watch on CRDs can use the `GetActiveCDI()` function to queue CDI CR.  The work of `ReconcileCDICRD` can then be moved to a callback function

Reply:
I'm having trouble getting this code to work. the operator lifecycle sdk only triggers the callbacks for resources from GetAllResources, but if I add CDI CRD to the list, cdi-operator loops and calls CDI CRD and orphan object.

[Code here](https://github.com/maya-r/containerized-data-importer/tree/v1alpha1-remove-cdi-crd2).

Reply:
I meant that you call `GetActiveCDI` here:  https://github.com/maya-r/containerized-data-importer/blob/9123579359b3c241d1c06d1b04a61d29b48ea3a2/pkg/operator/controller/cruft.go#L280-L287

That will trigger regular reconcile.  Then add a callback on any resource like cdi-deployment to do the work of checking/updating the CDI CRD

Reply:
I've written the code this way, but I can't use a callback like for the other CRDs. if I add it as a resource that operator-lifecycle-sdk monitors, it errors and fails to deploy CDI complaining about the "orphan" resource of CDI CRD (that isn't owned by the CDI deployment). It looks better, I hope that's good enough.

Reply:
Not sure what you were trying to do but operator-lifecycle-sdk can be totally oblivious.  Take a look at the `reconcileCreateRoute` callback.  It is a callback on deployments that is executed when cdi-deployment is passed.  But that is really just a mechanism to make sure that the callback only executes once per reconcile loop.  The callback has nothing to do with that cdi-deployment.  It manages a Route that operator-lifecycle-sdk does not care about.  We can do something similar for the CDI CR.

Reply:
Done, thanks for the tip